test/parallel/test-http2-client-destroy.js

```diff
@@ -71,19 +71,40 @@ const h2 = require('http2');
   );
 }
 
-// test destroy before connect
+// test destroy before client operations
 {
   const server = h2.createServer();
   server.listen(
     0,
     common.mustCall(() => {
       const client = h2.connect(`http://localhost:${server.address().port}`);
-
-      const req = client.request({ ':path': '/' });
+      const req = client.request();
       client.destroy();
 
       req.on('response', common.mustNotCall());
       req.resume();
+
+      const sessionError = {
+        type: Error,
+        code: 'ERR_HTTP2_INVALID_SESSION',
+        message: 'The session has been destroyed'
+      };
+
+      common.expectsError(() => client.request(), sessionError);
+      common.expectsError(() => client.settings({}), sessionError);
+      common.expectsError(() => client.priority(req, {}), sessionError);
+      common.expectsError(() => client.shutdown(), sessionError);
+
+      // Wait for setImmediate call from destroy() to complete
+      // so that state.destroyed is set to true
+      setTimeout(() => {
+        common.expectsError(() => client.request(), sessionError);
+        common.expectsError(() => client.settings({}), sessionError);
+        common.expectsError(() => client.priority(req, {}), sessionError);
+        common.expectsError(() => client.shutdown(), sessionError);
+        common.expectsError(() => client.rstStream(req), sessionError);
+      }, common.platformTimeout(100));
```

Comment:
Arbitrary timeouts are a code smell. Please find another way to solve whatever problem it is supposed to solve. Listening for events or even an unfortunate `setInterval()` for polling would be preferable.  If you must use `common.platformTimeout()` chances are this should be moved to `sequential`.

Reply:
Here's this chunk of code rewritten with a polling `setInterval()`:

```js
      const interval = setInterval(() => {
        if (client.destroyed) {
          clearInterval(interval);
          common.expectsError(() => client.request(), sessionError);
          common.expectsError(() => client.settings({}), sessionError);
          common.expectsError(() => client.priority(req, {}), sessionError);
          common.expectsError(() => client.shutdown(), sessionError);
          common.expectsError(() => client.rstStream(req), sessionError);          
        }
      }, 1);
```

Using that instead makes it reliable under load:

```console
$ tools/test.py -j 96 --repeat 192 test/parallel/test-http2-client-destroy.js 
[00:12|% 100|+ 192|-   0]: Done                              
$
```

Reply:
As far as I can tell, this never fails with `setImmediate`.

```js
Machine:nodejs apapirovski$ tools/test.py -j 96 --repeat 960 test/parallel/test-http2-client-destroy.js
[00:30|% 100|+ 960|-   0]: Done                              
```

I don't really understand why it was changed when the previous CI ran successfully.

Reply:
I too am also seeing 100% success with `setImmediate()`. Putting it back to that is probably the most elegant solution. CI failures seem unrelated to this PR.

(Speaking of Ci challenges: Anybody good with Jenkins and/or ansible and/or systems administration would probably make a great addition to the Build WG. If that sounds like you and you are a known quantity around these parts, maybe nominate yourself with an issue in https://github.com/nodejs/build? There are some talented folks on that team, but we probably could use many more active members.)

Reply:
Switched to `setImmediate()` in the new commit
@Trott Can you create a new issue for CI challenges and reference this PR?